bluepysnap/circuit_validation.py

```diff
@@ -381,13 +423,14 @@ def _check_nodes_population(nodes_dict, config):
                 return [fatal('Population {} of {} misses datasets {}'.
                               format(population_name, nodes_file, missing_datasets))]
             if len(groups) > 1:
-                m_errors = _check_multi_groups(population['node_group_id'],
-                                               population['node_group_index'], population)
-                if len(m_errors) > 0:
-                    return m_errors
+                errors += _check_multi_groups(population['node_group_id'],
+                                              population['node_group_index'], population)
+                if len(errors) > 0:
+                    return errors
```

Comment:
I think the general idea is to return immediately if the error is fatal one.  At least that's the idea I followed. In this case `_check_multi_groups` only has fatal errors so if there's any errors, I'll return it.

Reply:
It's a general rule in validation to return if there is a fatal error. Everything is good here. The only problem is that now it validates the spec that is not open-sourced while the project is open-sourced.

Reply:
Wouldn't make sense to return all the errors, unless it's not possible to proceed with the validation?
In this way the user would be able to fix all of them at once.

Reply:
I didn't check the exact situation here but a fatal error means that validation can't be continued because there is error in Sonata structure that does prevent further reading of data.